activerecord/lib/active_record/connection_adapters/connection_specification.rb

```diff
@@ -126,42 +159,35 @@ def initialize(configurations)
         #
         # Full hash Configuration.
         #
-        #   configurations = { "production" => { "host" => "localhost", "database" => "foo", "adapter" => "sqlite3" } }
-        #   Resolver.new(configurations).resolve(:production)
+        #   configurations = { "primary" => { "host" => "localhost", "database" => "foo", "adapter" => "sqlite3" } }
+        #   Resolver.new(configurations).resolve(:primary)
         #   # => { "host" => "localhost", "database" => "foo", "adapter" => "sqlite3"}
         #
         # Initialized with URL configuration strings.
         #
-        #   configurations = { "production" => "postgresql://localhost/foo" }
-        #   Resolver.new(configurations).resolve(:production)
+        #   configurations = { "primary" => "postgresql://localhost/foo" }
+        #   Resolver.new(configurations).resolve(:primary)
         #   # => { "host" => "localhost", "database" => "foo", "adapter" => "postgresql" }
         #
         def resolve(config)
           if config
             resolve_connection config
-          elsif env = ActiveRecord::ConnectionHandling::RAILS_ENV.call
-            resolve_symbol_connection env.to_sym
           else
             raise AdapterNotSpecified
           end
         end
 
-        # Expands each key in @configurations hash into fully resolved hash
-        def resolve_all
-          config = configurations.dup
-          config.each do |key, value|
-            config[key] = resolve(value) if value
-          end
-          config
-        end
-
         # Returns an instance of ConnectionSpecification for a given adapter.
-        # Accepts a hash one layer deep that contains all connection information.
+        # Accepts:
+        # - Hash: one layer deep Hash that contains all connection information
+        # - Symbol: a configuration name that will be looked-up
+        #           from the configurations Hash
+        # - String: a database url
         #
         # == Example
         #
-        #   config = { "production" => { "host" => "localhost", "database" => "foo", "adapter" => "sqlite3" } }
-        #   spec = Resolver.new(config).spec(:production)
+        #   config = { "host" => "localhost", "database" => "foo", "adapter" => "sqlite3" }
+        #   spec = Resolver.new({}).spec(config)
```

Comment:
This API looks weird to me. I'd expect the resolver to always be initialized with the config and resolve a spec.